assignments/assignment02.rb

```diff
@@ -0,0 +1,332 @@
+# ========================================================================================
+# Assignment 2
+# ========================================================================================
+
+# ========================================================================================
+#  Problem 1 - `to_sentence`
+
+# implement method `to_sentence`
+
+# creates an english string from array
+
+def to_sentence(ary)
+  # your implementation here
+  ary_len = ary.length
+  result_ary = []
+  i = 1;
+
+  #push first word from array to result string
+  result_ary << ary[0]
+
+  #if array is has more than one word 
+  if ary_len > 1
+    #iterate through ary except for the last one 
+    while i < ary_len-1
+      #push a comma and a space to result
+      result_ary << ", "
+      #push a word from array to result
+      result_ary << ary[i]
+      i += 1
+    end
+
+    #push and before last word
+    result_ary << " and "
+    #push last word to result
+    result_ary << ary[i]
+  end
+   
+  return result_ary.join
+  
+end
+
+# Your method should generate the following results:
+to_sentence []                       #=> ""
+to_sentence ["john"]                 #=> "john"
+to_sentence ["john", "paul"]         #=> "john and paul"
+to_sentence [1, "paul", 3, "ringo"]  #=> "1, paul, 3 and ringo"
+
+
+# ========================================================================================
+#  Problem 2 - `mean, median`
+
+# implement methods "mean", "median" on Array of numbers
+def mean(ary)
+  # get sum
+  sum = ary.reduce(0.0) {|item, acc| acc + item}
+  
+  # divide by the numbers of element
+  return sum/ary.count
+end
+
+def median(ary)
+  n = ary.count
+  m = nil
+
+  #sort
+  sorted_ary = ary.sort
+
+  #if ary has even number of element return average of element count/2 and count/2 
+  if(n % 2 == 0)
```

Comment:
in ruby, 'if' is an expression that returns a value. You don't need the local variable m, since the value of if will be the value of the last statement in either branch. You also don't need the "return m" at the end of the method, since the method would automatically return the value of the if expression, since it would be the last expression in the method.